browser-interface/src/index.ts

```diff
@@ -1 +1,89 @@
+export type UnityGame = {
+  Module: {
+    /** this handler can be overwritten, return true to stop error propagation */
+    errorHandler?: (message: string, filename: string, lineno: number) => boolean
+  }
+  SendMessage(object: string, method: string, args: number | string): void
+  SetFullscreen(): void
+  Quit(): Promise<void>
+}
+
+export type RendererOptions = {
+  canvas: HTMLCanvasElement
+
+  onProgress?: (progress: number) => void
+  onSuccess?: (unityInstance: any) => void
+  onError?: (error: any) => void
+  /** Legacy messaging system */
+  onMessageLegacy: (type: string, payload: string) => void
+  /** used to append a ?v={} to the URL. Useful to debug cache issues */
+  versionQueryParam?: string
+  /** baseUrl where all the assets are deployed */
+  baseUrl: string
+}
+
+export type DecentralandRendererInstance = {
+  /**
+   * Signal sent by unity after it started correctly
+   * it is a promise, that makes it awaitable.
+   * The content of the resolved promise is an empty object to
+   * enable future extensions.
+   */
+  engineStartedFuture: Promise<{}>
+
+  // soon there will be more protocol functions here https://github.com/decentraland/renderer-protocol
+  // and originalUnity will be deprecated to decouple the kernel from unity's impl internals
+  originalUnity: UnityGame
+}
+
+export async function initializeWebRenderer(options: RendererOptions): Promise<DecentralandRendererInstance> {
+  const rendererVersion = options.versionQueryParam || performance.now()
+  const { canvas, baseUrl, onProgress, onSuccess, onError, onMessageLegacy } = options
+  const resolveWithBaseUrl = (file: string) => new URL(file + "?v=" + rendererVersion, baseUrl).toString()
+
+  const config = {
+    dataUrl: resolveWithBaseUrl("unity.data.unityweb"),
+    frameworkUrl: resolveWithBaseUrl("unity.framework.js.unityweb"),
+    codeUrl: resolveWithBaseUrl("unity.wasm.unityweb"),
+    streamingAssetsUrl: resolveWithBaseUrl("StreamingAssets"),
+    companyName: "Decentraland",
+    productName: "Decentraland World Client",
+    productVersion: "0.1",
+  }
+
+  const engineStartedFuture = future<{}>()
+
+  // The namespace DCL is exposed to global because the unity template uses it to send the messages
+  // @see https://github.com/decentraland/unity-renderer/blob/bc2bf1ee0d685132c85606055e592bac038b3471/unity-renderer/Assets/Plugins/JSFunctions.jslib#L6-L29
+  ;(globalThis as any)["DCL"] = {
+    // This function get's called by the engine
+    EngineStarted() {
+      engineStartedFuture.resolve({})
+    },
+
+    // This function is called from the unity renderer to send messages back to the scenes
+    MessageFromEngine(type: string, jsonEncodedMessage: string) {
+      onMessageLegacy(type, jsonEncodedMessage)
+    },
+  }
+
+  const originalUnity = await createUnityInstance(canvas, config, onProgress, onSuccess, onError)
+
+  // TODO: replace originalUnity.errorHandler with a version that redirects errors to -> onError
```

Comment:
Should we add an action point for this? I don't understand well enough the context.

Reply:
It is a slow migration, no need to do it on this PR, the comment is there to generate awareness that `onError` should be the unique place where we will handle errors from unity-renderer, to remove the specialized code from `kernel`